client/app/components/OnboardingTopBar/OnboardingTopBar.js

```diff
@@ -4,22 +4,51 @@ import { div, span, a } from 'r-dom';
 
 import css from './OnboardingTopBar.scss';
 
-const next = function next(nextStep, guideRoot, t) {
-  return t(nextStep) ? {
-    title: t(nextStep),
-    link: `${guideRoot}/${nextStep}`,
-  } : null;
+const next = function next(nextStep, guideRoot) {
+  switch (nextStep) {
+  case "slogan_and_description":
+    return {
+      title: I18n.t("admin.onboarding.topbar.slogan_and_description"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "cover_photo":
+    return {
+      title: I18n.t("admin.onboarding.topbar.cover_photo"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "filter":
+    return {
+      title: I18n.t("admin.onboarding.topbar.filter"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "paypal":
+    return {
+      title: I18n.t("admin.onboarding.topbar.paypal"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "listing":
+    return {
+      title: I18n.t("admin.onboarding.topbar.listing"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "invitation":
+    return {
+      title: I18n.t("admin.onboarding.topbar.invitation"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  default:
+    return null;
+  }
```

Comment:
We are now tightly coupling frontend and backend code. :(
All client devs (web apps, mobile apps etc.) need to understand Rails-side translation context and they need to write their code in a way that Rails-side changes can be grepped from their code-base.

Related:
I also think that we should think about hard-coding all the needed locales to PropTypes of Page-component, 
```
Page.propTypes = {
  translations: PropTypes.objectOf(PropTypes.shape({
    admin: PropTypes.objectOf(PropTypes.shape({
      onboarding: PropTypes.objectOf(PropTypes.shape({
        topbar: PropTypes.objectOf(PropTypes.shape({
          slogan_and_description: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    }).isRequired,
  }).isRequired,
};
```
Looks a bit heavy, but that way we get an alert if something is missing while rendering.

That method doesn't say that we should pass translations as props from haml/erb - in startup / registration step we could figure out what translations we actually need and pass them as props from there to actual Page component.